llama-index-core/tests/indices/test_prompt_helper.py

```diff
@@ -194,3 +195,35 @@ def test_get_biggest_prompt() -> None:
     biggest_prompt = get_biggest_prompt([prompt1, prompt2, prompt3])
 
     assert biggest_prompt == prompt2
+
+
+def test_json_in_prompt() -> None:
+    """Test that a JSON object in the prompt doesn't cause an error."""
+    # test with normal prompt
+    prompt = PromptTemplate(
+        'This is the prompt {text} but it also has {"json": "in it"}'
```

Comment:
Can you also test some stuff like the original failure:
```python
s = f"This is a {key}"
regs = {key:val for d.items()
```

Reply:
Sorry, can you flesh that out a bit more -- I'm not sure what the string or partial_format args should be exactly

Reply:
The original issue I discovered was that string_format() couldn't handle `{key:` since it is expecting a string formatter (i.e. `f"{some_number:#x}"`. But in this case, it was just a dictionary comprehension

Reply:
Concretely, what should I be testing here then?

```python
prompt = PromptTemplate('Here's a {var}, here's some other stuff: {key: val for val in d.items()}')
prompt.partial_format(var="hello_world")
```

Reply:
Added something like the above in any case

Reply:
Thank you!